[xyT-T] iP

xyT-T opened the pull request:
# Slay!

> "Work. Play. Slay."

Slay your tasks and slay your day with the help of _Slay!_ Slay helps you to:

- [x] manage tasks
- [x] remember deadlines
- [ ] organize your work

It's,

- text-based
- easy to use
- fast
- ~~cute~~ SUPER CUTE to interact with :D

All you need to do is,

1. Download it from [here](https://github.com/xyT-T/ip/releases).
2. Do a double-click.
3. Add your tasks.
4. Have fun with it and **SLAY YOUR DAY** 🦖

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Slay {
    public static void main(String[] args) {
        new Slay().run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

xyT-T committed:
Level-0

xyT-T committed:
Level-2 input accepted

xyT-T committed:
Level-2 add to list

xyT-T committed:
Level-3 mark/unmark tasks

xyT-T committed:
Level-4 different task types

xyT-T committed:
A-TextUiTesting

xyT-T committed:
Level-6 delete tasks

xyT-T committed:
Level-5 exceptions

xyT-T committed:
Level-5 exceptions

xyT-T committed:
Level-7

xyT-T committed:
Merge branch 'branch-Level-7'

xyT-T committed:
Level-8

xyT-T committed:
Merge branch 'branch-Level-8'

trgao commented on src/main/java/Tong.java:
```diff
@@ -0,0 +1,132 @@
+import java.util.Arrays;
+
+public class Tong {
+    public static void main(String[] args) {
```
Perhaps you should abstract out some methods so that the main method is not so long and difficult to read?

trgao commented on src/main/java/Tong.java:
```diff
@@ -0,0 +1,132 @@
+                int order;
+
+                switch (command) {
+                    case "list":
```
Perhaps you can remove the indentation for the case statements? (switch and case are at the same level of indentation)

trgao reviewed (commented):
Perhaps you can abstract out the code into different classes, but otherwise LGTM.

xyT-T committed:
A-MoreOOP

xyT-T committed:
Merge branch 'branch-A-MoreOOP'

xyT-T committed:
A-Packages organized classes into packages

xyT-T committed:
Merge branch 'master' into add-gradle-support

Song-Mengfei commented on src/main/java/tong/Parser.java:
```diff
@@ -0,0 +1,164 @@
+package tong;
+
+import tong.command.*;
```
It might be better to import every classes seperately

xyT-T committed:
A-Gradle

xyT-T committed:
Merge branch 'add-gradle-support'

Song-Mengfei commented on src/main/java/tong/Parser.java:
```diff
@@ -0,0 +1,164 @@
+        String description = matcher.group("description");
+
+        switch (type) {
+            case ToDo.TYPE:
```
There is no indentation for case clauses according to our coding standard 😥
I didn't realise it as well

Song-Mengfei commented on src/main/java/tong/Ui.java:
```diff
@@ -0,0 +1,53 @@
+    /**
+     * Shows the result of a command execution to the user. Includes additional formatting to demarcate different
+     * command execution segments.
+     */
```
I think the method parameters and return value should be involved in JavaDoc.
Such as:

/**
* This method is for ...
*
* @param name ...
* @return returnValue ...
*/

Song-Mengfei commented on src/main/java/tong/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,43 @@
+            taskList.unmarkTask(toUnmark);
+            return new CommandResult(String.format(MESSAGE_UNMARK_TASK_SUCCESS, toUnmark));
+        } catch (TaskNotFoundException e) {
+            return new CommandResult("You don't seem to have so many tasks. " +
```
It might be more clear if you break before the operator "+"

Song-Mengfei commented on src/main/java/tong/Parser.java:
```diff
@@ -0,0 +1,164 @@
+                    + "/from\\s+"
+                    + "(?<start>\\S+\\s+\\S+)\\s+"
+                    + "/to\\s+"
+                    + "(?<end>\\S+\\s+\\S+)$");
```
I like the way you used Pattern and final value 😀

Song-Mengfei reviewed (commented):
Good job! I have learnt a lot!

xyT-T committed:
A-JUnit

xyT-T committed:
Merge branch 'branch-A-JUnit'

xyT-T committed:
A-JavaDoc

xyT-T committed:
A-CodingStandard

xyT-T committed:
Level-9

xyT-T committed:
Merge branch 'branch-A-JavaDoc'

xyT-T committed:
Merge branch 'branch-A-CodingStandard'

xyT-T committed:
Merge branch 'branch-Level-9'

xyT-T committed:
Level-10

xyT-T committed:
Level-10 round user image

xyT-T committed:
Merge branch 'branch-Level-10'

xyT-T committed:
Name magic strings

There are many magic strings without explanation in the code.

Let's name such strings through constant declaration to provide meanings
for them.

Naming the magic strings makes the code more readily understood.

xyT-T committed:
Merge pull request #1 from xyT-T/branch-A-CodeQuality

Name magic strings

xyT-T committed:
Parser class: Shorten prepareAdd() method

The prepareAdd() method is very long due to the complexity in parsing
the command arguments of deadline and event tasks. This makes the code
less readable.

Extract the implementation of parsing into two seperate methods and
shorten prepareAdd() method.

xyT-T committed:
Merge pull request #3 from xyT-T/branch-A-CodeQuality

Parser class: Shorten prepareAdd() method

xyT-T committed:
Use assertions

xyT-T committed:
Merge branch 'master' into branch-A-Assertions

xyT-T committed:
Merge pull request #4 from xyT-T/branch-A-Assertions

Use assertions

xyT-T committed:
Add extention tag

xyT-T committed:
Merge branch 'branch-C-Tagging'

xyT-T committed:
Merge PR and add extension tag

xyT-T committed:
GUI bug fix: Show full message

xyT-T committed:
Merge branch 'branch-A-BetterGui'

xyT-T committed:
Add more tests

xyT-T committed:
Merge branch 'branch-A-MoreTesting'

xyT-T committed:
Handle empty argument exception

xyT-T committed:
Merge branch 'branch-A-MoreErrorHandling'

xyT-T committed:
Show welcome message

xyT-T committed:
GUI: Add title

xyT-T committed:
Update Ui.png

xyT-T committed:
Update README.md

xyT-T committed:
Update README.md

xyT-T committed:
Update README.md

xyT-T committed:
Add more JavaDoc

xyT-T committed:
Handle invalid input date and time

xyT-T committed:
Bug fix: Exit correctly

xyT-T committed:
Bug fix: Add find command into help message
